runtime/src/main/java/io/quarkus/vault/runtime/VaultKvManager.java

```diff
@@ -33,23 +41,48 @@ public static VaultKvManager privateClientManager(@Private VaultClient vaultClie
                 vaultInternalKvV2SecretEngine);
     }
 
-    private VaultClient vaultClient;
-    private VaultAuthManager vaultAuthManager;
-    private VaultConfigHolder vaultConfigHolder;
-    private VaultInternalKvV1SecretEngine vaultInternalKvV1SecretEngine;
-    private VaultInternalKvV2SecretEngine vaultInternalKvV2SecretEngine;
+    @Inject
+    public VaultKvManager(VaultClient vaultClient,
+            VaultAuthManager vaultAuthManager,
+            VaultConfigHolder vaultConfigHolder,
+            VaultInternalKvV1SecretEngine vaultInternalKvV1SecretEngine,
+            VaultInternalKvV2SecretEngine vaultInternalKvV2SecretEngine) {
+        this.vaultClient = vaultClient;
+        this.vaultAuthManager = vaultAuthManager;
+        this.vaultConfigHolder = vaultConfigHolder;
+        this.vaultInternalKvV1SecretEngine = vaultInternalKvV1SecretEngine;
+        this.vaultInternalKvV2SecretEngine = vaultInternalKvV2SecretEngine;
+        this.mount = this.vaultConfigHolder.getVaultRuntimeConfig().kvSecretEngineMountPath();
+    }
 
-    public VaultKvManager(VaultClient vaultClient, VaultAuthManager vaultAuthManager,
+    public VaultKvManager(String mount,
+            VaultClient vaultClient,
+            VaultAuthManager vaultAuthManager,
             VaultConfigHolder vaultConfigHolder,
             VaultInternalKvV1SecretEngine vaultInternalKvV1SecretEngine,
             VaultInternalKvV2SecretEngine vaultInternalKvV2SecretEngine) {
+        this.mount = mount;
         this.vaultClient = vaultClient;
         this.vaultAuthManager = vaultAuthManager;
         this.vaultConfigHolder = vaultConfigHolder;
         this.vaultInternalKvV1SecretEngine = vaultInternalKvV1SecretEngine;
         this.vaultInternalKvV2SecretEngine = vaultInternalKvV2SecretEngine;
     }
 
+    public static VaultKvManager of(String mount,
```

Comment:
Why add a static constructor? The constructor is only ever called by CDI or the engine factory?